File: src/Complex/Prog7/Forcedresponse/Forcedresponse.java
package Complex.Prog7.Forcedresponse;

import java.applet.Applet;
import java.awt.BorderLayout;
import java.awt.Image;
import java.awt.Label;
import java.awt.Panel;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.Date;
import javax.swing.Timer;

@SuppressWarnings("serial")
public class Forcedresponse extends Applet implements ActionListener {
   static final double kMinL = 0.5;
   static final double kMaxL = 2.0;
   static final double kMinD = 0.05;
   static final double kMaxD = 0.2;
   static final double kMinm = 5.0;
   static final double kMaxm = 100.0;
   static final double kMink = 1.5E8;
   static final double kMaxk = 1.0E10;
   static final String kRunString = "Run";
   CFramePanel mFramePanel;
   CFrameAnimation mAnimFrame;
   Timer mTimer = null;
   long mLastTime = 0;

   public void init() {
      this.setLayout(new BorderLayout());
      Panel buttonPanel = new Panel();
      this.add("North", buttonPanel);
      Label copyright = new Label("  This program is copyright, you need permission to use it. © 2004 B J Stone, bjs@mech.uwa.edu.au.");
      this.add("South", copyright);
      this.mFramePanel = new CFramePanel(this);
      this.add("Center", this.mFramePanel);
      this.mAnimFrame = new CFrameAnimation(this.mFramePanel, 0, 0, this.getSize().width, this.getSize().height);
      CFrameSmallControl LControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 1, 450, 310, 0.5, 1.0, 2.0, "m");
      LControl.SetLabelOffset(20, 10);
      LControl.SetTextEditable(true);
      Image LImage = this.getImage(this.getCodeBase(), "L.gif");
      new CFramePicture(this.mFramePanel, 475, 325, 0, 0, LImage, false);
      CFrameSmallControl DControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 2, 530, 310, 0.05, 0.1, 0.2, "m");
      DControl.SetLabelOffset(20, 10);
      DControl.SetTextEditable(true);
      Image DImage = this.getImage(this.getCodeBase(), "D.gif");
      new CFramePicture(this.mFramePanel, 555, 325, 0, 0, DImage, false);
      CFrameSmallControl mControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 3, 610, 310, 5.0, 20.0, 100.0, "kg");
      mControl.SetLabelOffset(20, 10);
      mControl.SetTextEditable(true);
      Image mImage = this.getImage(this.getCodeBase(), "m.gif");
      new CFramePicture(this.mFramePanel, 635, 325, 0, 0, mImage, false);
      CFrameSmallControl kControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 4, 690, 310, 1.5E8, 1.0E9, 1.0E10, "N/m");
      kControl.SetLabelOffset(20, 10);
      kControl.SetTextEditable(true);
      Image kImage = this.getImage(this.getCodeBase(), "k.gif");
      new CFramePicture(this.mFramePanel, 715, 325, 0, 0, kImage, false);
      Image DiagramImage = this.getImage(this.getCodeBase(), "Diagram.gif");
      new CFramePicture(this.mFramePanel, 470, 30, 0, 0, DiagramImage, false);
      Image XaxisImage = this.getImage(this.getCodeBase(), "Xaxis.gif");
      new CFramePicture(this.mFramePanel, 87, 185, 0, 0, XaxisImage, false);
      Image YaxisImage = this.getImage(this.getCodeBase(), "Yaxis.gif");
      new CFramePicture(this.mFramePanel, 0, 3, 0, 0, YaxisImage, false);
      this.validate();
   }

   public void start() {
      this.mTimer = new Timer(20, this);
      this.mTimer.start();
   }

   public void actionPerformed(ActionEvent e) {
      if (this.mTimer != null) {
         Date ddd = new Date();
         long thisTime = ddd.getTime();
         
         if (this.mLastTime == 0) {
            this.mLastTime = thisTime;
         }
         double realSeconds = (thisTime - this.mLastTime) / 1000.0;
         this.mLastTime = thisTime;
         
         //this.mAnimFrame.ControlMessage(0, realSeconds);
      }
   }
}

File: site/swingjs/j2s/Complex/Prog7/Forcedresponse/Forcedresponse.js
(function(){var P$=Clazz.newPackage("Complex.Prog7.Forcedresponse"),I$=[[0,'java.awt.BorderLayout','java.awt.Panel','java.awt.Label','Complex.Prog7.Forcedresponse.CFramePanel','Complex.Prog7.Forcedresponse.CFrameAnimation','Complex.Prog7.Forcedresponse.CFrameSmallControl','Complex.Prog7.Forcedresponse.CFramePicture','javax.swing.Timer']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Forcedresponse", null, 'java.applet.Applet', 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.mTimer=null;
this.mLastTime=0;
},1);

C$.$fields$=[['J',['mLastTime'],'O',['mFramePanel','Complex.Prog7.Forcedresponse.CFramePanel','mAnimFrame','Complex.Prog7.Forcedresponse.CFrameAnimation','mTimer','javax.swing.Timer']]]

Clazz.newMeth(C$, ['init$','init'],  function () {
this.setLayout$java_awt_LayoutManager(Clazz.new_($I$(1,1)));
var buttonPanel=Clazz.new_($I$(2,1));
this.add$S$java_awt_Component("North", buttonPanel);
var copyright=Clazz.new_($I$(3,1).c$$S,["  This program is copyright, you need permission to use it. \u00a9 2004 B J Stone, bjs@mech.uwa.edu.au."]);
this.add$S$java_awt_Component("South", copyright);
this.mFramePanel=Clazz.new_($I$(4,1).c$$java_applet_Applet,[this]);
this.add$S$java_awt_Component("Center", this.mFramePanel);
this.mAnimFrame=Clazz.new_([this.mFramePanel, 0, 0, this.getSize$().width, this.getSize$().height],$I$(5,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I);
var LControl=Clazz.new_($I$(6,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$Complex_Prog7_Forcedresponse_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 1, 450, 310, 0.5, 1.0, 2.0, "m"]);
LControl.SetLabelOffset$I$I(20, 10);
LControl.SetTextEditable$Z(true);
var LImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "L.gif");
Clazz.new_($I$(7,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 475, 325, 0, 0, LImage, false]);
var DControl=Clazz.new_($I$(6,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$Complex_Prog7_Forcedresponse_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 2, 530, 310, 0.05, 0.1, 0.2, "m"]);
DControl.SetLabelOffset$I$I(20, 10);
DControl.SetTextEditable$Z(true);
var DImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "D.gif");
Clazz.new_($I$(7,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 555, 325, 0, 0, DImage, false]);
var mControl=Clazz.new_($I$(6,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$Complex_Prog7_Forcedresponse_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 3, 610, 310, 5.0, 20.0, 100.0, "kg"]);
mControl.SetLabelOffset$I$I(20, 10);
mControl.SetTextEditable$Z(true);
var mImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "m.gif");
Clazz.new_($I$(7,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 635, 325, 0, 0, mImage, false]);
var kControl=Clazz.new_($I$(6,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$Complex_Prog7_Forcedresponse_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 4, 690, 310, 1.5E8, 1.0E9, 1.0E10, "N/m"]);
kControl.SetLabelOffset$I$I(20, 10);
kControl.SetTextEditable$Z(true);
var kImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "k.gif");
Clazz.new_($I$(7,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 715, 325, 0, 0, kImage, false]);
var DiagramImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Diagram.gif");
Clazz.new_($I$(7,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 470, 30, 0, 0, DiagramImage, false]);
var XaxisImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Xaxis.gif");
Clazz.new_($I$(7,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 87, 185, 0, 0, XaxisImage, false]);
var YaxisImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Yaxis.gif");
Clazz.new_($I$(7,1).c$$Complex_Prog7_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 0, 3, 0, 0, YaxisImage, false]);
this.validate$();
});

Clazz.newMeth(C$, ['start$','start'],  function () {
this.mTimer=Clazz.new_($I$(8,1).c$$I$java_awt_event_ActionListener,[20, this]);
this.mTimer.start$();
});

Clazz.newMeth(C$, ['actionPerformed$java_awt_event_ActionEvent','actionPerformed'],  function (e) {
if (this.mTimer != null ) {
var ddd=Clazz.new_(java.util.Date);
var thisTime=ddd.getTime$();
if (Long.$eq(this.mLastTime,0 )) {
this.mLastTime=thisTime;
}this.mLastTime=thisTime;
}});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:48 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
